docs/user-guide.rst

```diff
@@ -0,0 +1,406 @@
+:attr:`~response.HTTPResponse.header` attributes::
+
+    >>> r = http.request('GET', 'http://httpbin.org/ip')
+    >>> r.status
+    200
+    >>> r.data
+    b'{\n  "origin": "104.232.115.37"\n}\n'
+    >>> r.headers
+    HTTPHeaderDict({'Content-Length': '33', ...})
+
+JSON content
+~~~~~~~~~~~~
+
+JSON content can be loaded by decoding and deserializing the
+:attr:`~response.HTTPResponse.data` attribute of the request::
+
+    >>> import json
+    >>> r = http.request('GET', 'http://httpbin.org/ip')
+    >>> json.loads(r.data.decode('utf-8'))
+    {'origin': '127.0.0.1'}
+
+Binary content
+~~~~~~~~~~~~~~
+
+The :attr:`~response.HTTPResponse.data` attribute of the response is always set
+to a byte string representing the response content::
+
+    >>> r = http.request('GET', 'http://httpbin.org/bytes/8')
+    >>> r.data
+    b'\xaa\xa5H?\x95\xe9\x9b\x11'
+
+.. note:: For larger responses, it's sometimes better to :ref:`stream <stream>`
+    the response.
+
+.. _request_data:
+
+Request data
+------------
+
+Query parameters
+~~~~~~~~~~~~~~~~
+
+For ``GET``, ``HEAD``, and ``DELETE`` requests, you can simply pass the
+arguments as a dictionary in the ``fields`` argument to
+:meth:`~poolmanager.PoolManager.request`::
+
+    >>> r = http.request(
+    ...     'GET',
+    ...     'http://httpbin.org/get',
+    ...     fields={'arg': 'value'})
+    >>> json.loads(r.data.decode('utf-8'))['args']
+    {'arg': 'value'}
+
+For ``POST`` and ``PUT`` requests, you need to manually encode query parameters
+in the URL::
+
+    >>> from urllib.parse import urlencode
+    >>> encoded_args = urlencode({'arg': 'value'})
+    >>> url = 'http://httpbin.org/post?' + encoded_args
+    >>> r = http.request('POST', url)
+    >>> json.loads(r.data.decode('utf-8'))['args']
+    {'arg': 'value'}
+
+Headers
+~~~~~~~
+
+You can specify headers as a dictionary in the ``headers`` argument in :meth:`~poolmanager.PoolManager.request`::
+
+    >>> r = http.request(
+    ...     'GET',
+    ...     'http://httpbin.org/headers',
+    ...     headers={
+    ...         'X-Something': 'value'
+    ...     })
+    >>> json.loads(r.data.decode('utf-8'))['headers']
+    {'X-Something': 'value', ...}
+
+
+.. _form_data:
+
+Form data
+~~~~~~~~~
+
+For ``PUT`` and ``POST`` requests, urllib3 will automatically form-encode the
+dictionary in the ``fields`` argument provided to
+:meth:`~poolmanager.PoolManager.request`::
+
+    >>> r = http.request(
+    ...     'POST',
+    ...     'http://httpbin.org/post',
+    ...     fields={'field': 'value'})
+    >>> json.loads(r.data.decode('utf-8'))['form']
+    {'field': 'value'}
+
+JSON
+~~~~
+
+You can sent JSON a request by specifying the encoded data as the ``body``
+argument and setting the ``Content-Type`` header when calling 
+:meth:`~poolmanager.PoolManager.request`::
+
+    >>> import json
+    >>> data = {'attribute': 'value'}
+    >>> encoded_data = json.dumps(data)
```

Comment:
Yeah, it generally will, because we do have logic that encodes Unicode strings to bytes, but there are some edge cases where that can cause misbehavior, and we do expect bytes for request bodies.